src/extension/provider/statefulAuth.ts

```diff
@@ -248,6 +237,71 @@ export class StatefulAuthProvider implements AuthenticationProvider, Disposable
     this.#disposables.forEach((d) => d.dispose())
   }
 
+  public async loadFromSecrets() {
+    const secretsFile = '/etc/secrets/playground-dev'
+    const secretsUri = Uri.parse(secretsFile)
+    const hasSecretsFile = await workspace.fs.stat(secretsUri).then(
+      () => true,
+      () => false,
+    )
+
+    if (!hasSecretsFile) {
+      logger.info('No secrets file found, skipping load from secrets')
+      return
+    }
+
+    const rawToken = await workspace.fs.readFile(secretsUri)
+    if (!rawToken?.length) {
+      logger.error('Failed to read secrets file')
+      return
+    }
+
+    const token = new TextDecoder().decode(rawToken).trim()
+    const jwtDecoded = jwt.decode(token) as DecodedToken
+    if (!jwtDecoded) {
+      logger.error('Failed to decode JWT token')
+      return
+    }
+
+    const { exp, scope } = jwtDecoded
+    if (!exp || !scope) {
+      logger.error('Invalid JWT token')
+      return
+    }
+
+    try {
```

Comment:
Let's break out the code into a private/protected method that makes the authentication decision and name the method accordingly.

Having less detail right here and calling into subroutines with distinct names will clarify that.